hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/CryptoApproveAllowanceSuite.java

```diff
@@ -443,11 +443,15 @@ private HapiSpec approveForAllSpenderCanDelegateOnNFT() {
                                         OWNER, NON_FUNGIBLE_TOKEN, delegatingSpender, newSpender, false, List.of(2L))
                                 .signedBy(DEFAULT_PAYER, newSpender)
                                 .hasPrecheck(DELEGATING_SPENDER_DOES_NOT_HAVE_APPROVE_FOR_ALL),
+                        // NOTE: This test in mono-service was failing in pre-check before signing requirements
+                        // (the signing requirements are wrong too).
+                        // In new code since we moved these checks to handle, it failed at signature verification.
+                        // To make it fail with same status code, we need to sign with owner.
                         cryptoApproveAllowance()
                                 .payingWith(DEFAULT_PAYER)
                                 .addDelegatedNftAllowance(
                                         OWNER, NON_FUNGIBLE_TOKEN, newSpender, delegatingSpender, true, List.of())
-                                .signedBy(DEFAULT_PAYER, delegatingSpender)
+                                .signedBy(DEFAULT_PAYER, OWNER)
```

Comment:
With this change, will it pass in mono code test too?

Reply:
Yes Iris, it will pass, because in mono-service the precheck still fails